File: test_script/bench/bench_delegate.py
import asyncio

def yield_fn():
    for i in range(3_000_000):
        yield i

def delegate_test():
    yield from yield_fn()
    yield from yield_fn()
    yield from yield_fn()

def main():
    delegate_obj = delegate_test()
    for i in delegate_obj:
        pass

if __name__ == "__main__":
    main()

File: test_script/bench/bench_delegate.fs
@async
fn yield_fn() {
    for i in 0..3000000 {
        i.yield
    }
}

@async
fn delegate_test() {
    yield_fn().delegate

    yield_fn().delegate

    yield_fn().delegate
}

delegate_obj = delegate_test()

for i in delegate_obj {
    #println(i)
}

dump(delegate_obj)